util/helpers.py

```diff
@@ -32,3 +57,53 @@ def check_ctcdecoder_version():
         sys.exit(1)
 
     return rv
+
+
+class LimitingPool:
```

Comment:
Why is this needed?

Reply:
`subprocess.Pool.imap` turned out to be unbound when it comes to processing items ahead of consuming them. This led to memory flooding, as sample contents are kept in memory.

Reply:
Can you please add this as a comment on the class?